code-review.js

```diff
@@ -1,60 +1,119 @@
+ *
+ * @class
  */
-function PriorityQueue(size) {
-	this.store = {};	// keys are priorities, values are arrays of elements
+function PriorityQueue() {
+	this.queue = {};	// keys are priorities, values are arrays of elements
 	this.count = 0;
 
-	// adds an item
-	// priority must be an integer (higher value has higher priority)
+
+	/**
+	 * Adds an item
+	 * priority must be a positive integer (higher value has higher priority)
+	 * 
+	 * @param {*} value - The Item we are adding to the queue
+	 * @param {Number} priority - Positive integer
+	 */
 	this.add = function(value, priority) {
-		if (this.store[priority] == undefined)
-			this.store[priority] = [];
+		if (this.queue[priority] == undefined) {
+			this.queue[priority] = [];
+		}
 
-		this.store[priority].push(value);
+		this.queue[priority].push(value);
 		this.count++;
 	};
 
-	// returns the oldest-added value with the highest priority
-	this.Pop = function() {
-		maxKey = Math.max(Object.keys(this.store));
+	/**
+	 * Returns the oldest-added value with the highest priority
+	 * @returns {*} - value removed
+	 */
+	this.pop = function() {
+		if (this.length() === 0) {
+			return false;
+		}
+		var highestPriority = Math.max.apply(null, this.getAllPriorities());
+		var poppedValue = this.queue[highestPriority].shift();
 		this.count--;
-		return this.store[maxKey].shift();
+		this.removeEmptyPriority(highestPriority);
+
+		return poppedValue
 	};
 
+	/**
+	 * Returns the number of elements in queue
+	 * @returns {Number} 
+	 */
 	this.length = function() {
 		return this.count;
-	}
-}
+	};
+};
 
-PriorityQueue.prototype.get_all_priorities = function() {
-	return Object.keys(this.store);
-}
+/**
+ * Returns all the priorities in descending order
+ * @returns {Array}
+ */
+PriorityQueue.prototype.getAllPriorities = function() {
+	return Object.keys(this.queue).sort(function(a, b){
+		return b - a;
+	});
+};
 
-// iterates through all the queue elements in priority-then-FIFO order
-PriorityQueue.prototype.forEach = function(callback) {
-	var keys = Object.keys(this.store).sort();
+/**
+ * Iterates through all the queue elements in priority-then-FIFO order
+ * 
+ * @param {function} callback - Callback function to be used on every element
+ */
+PriorityQueue.prototype.priorityForEach = function(callback) {
+	var allPriorities = this.getAllPriorities();
+	allPriorities.forEach(function(priority) {
+		for (var valueIndex = 0; valueIndex < this.queue[priority].length; valueIndex++) {
+			callback(this.queue[priority][valueIndex]);
+		}
+	}, this);
+};
 
-	for (var a = keys.length; a > 0; a--) {
-		for (var b = 0; b < this.store[a].length; b++)
-			callback(this.store[a][b]);
+/**
+ * Tests if a priority's array is empty and deletes the priority if not
+ * 
+ * @param {Number} priority - Priority to be tested
+ */
+PriorityQueue.prototype.removeEmptyPriority = function(priority) {
+	if (this.queue[priority].length === 0){
+		delete this.queue[priority];

```

Comment:
I didn't want to add any methods to the class and start overthinking this too much, but felt it was necessary for the implementation. Without this `pop()` would only be affective once and `changePriority()` could cause the same affect. I also didn't think it made sense having key's without any elements.